source/common/upstream/load_balancer_impl.cc

```diff
@@ -232,6 +233,11 @@ void LoadBalancerBase::recalculatePerPriorityState(uint32_t priority,
                                 per_priority_load.healthy_priority_load_.get().end(), 0) +
                     std::accumulate(per_priority_load.degraded_priority_load_.get().begin(),
                                     per_priority_load.degraded_priority_load_.get().end(), 0));
+
+  total_healthy_hosts = 0;
```

Comment:
This function contains an early return on line 195; total_healthy_hosts is not initialized on early return.